.github/workflows/publish-website.yml

```diff
@@ -0,0 +1,59 @@
+name: Publishing Documentation Site
+
+on:
+  workflow_dispatch:
```

Comment:
Do we need `workflow_dispatch`? Not sure when we'd need to manually publish the same commits that have been published already by the push?

Reply:
Sometimes it is handy to publish it manually - for example when we enable the GitHub actions based GitHub pages, and then want to trigger this workflow. 

Once we include the nightly scheduled run, we might also want to trigger it manually if we see the need to run it during the day if nightly is not enough.